pkg/cmd/server/origin/controller/interfaces.go

```diff
@@ -50,12 +51,20 @@ type ControllerClientBuilder interface {
 	controller.ControllerClientBuilder
 	KubeInternalClient(name string) (kclientsetinternal.Interface, error)
 	KubeInternalClientOrDie(name string) kclientsetinternal.Interface
+
+	// Legacy OpenShift client (pkg/client)
 	DeprecatedOpenshiftClient(name string) (osclient.Interface, error)
 	DeprecatedOpenshiftClientOrDie(name string) osclient.Interface
-	OpenshiftTemplateClient(name string) (templateclient.Interface, error)
-	OpenshiftTemplateClientOrDie(name string) templateclient.Interface
-	OpenshiftImageClient(name string) (imageclient.Interface, error)
-	OpenshiftImageClientOrDie(name string) imageclient.Interface
+
+	// OpenShift clients based on generated internal clientsets
+	OpenshiftInternalTemplateClient(name string) (templateclient.Interface, error)
+	OpenshiftInternalTemplateClientOrDie(name string) templateclient.Interface
+
+	OpenshiftInternalImageClient(name string) (imageclientinternal.Interface, error)
```

Comment:
@deads2k @soltysh minor, but I like this naming scheme better:

OpenShift[Internal|External]<GroupName>Client 

It matches the kube client scheme and allow us to add external clients in future...

Reply:
> @deads2k @soltysh minor, but I like this naming scheme better:
> 
> OpenShift[Internal|External]Client
> 
> It matches the kube client scheme and allow us to add external clients in future...

ok

Reply:
On a side not, I still wonder if we need both `*Client` and `*ClientOrDie` if we use the latter in all places that I've searched for.

Reply:
@soltysh the apiserver use the *Client version to report errors properly, we need both.